### PR TITLE
Improvement/cldsrv 548 flag update oplog

### PR DESCRIPTION
**Pull request by benzekrimaha**
This PR aims to introduce a flag that will be used by Arsenal to introduce a hack to write the MD twice in the oplog: one "deleted: true" copy of the previous MD, followed by the expected update with the new metadata. 

Issue: CLDSRV-548

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   "dependencies": {
     "@azure/storage-blob": "^12.12.0",
     "@hapi/joi": "^17.1.0",
-    "arsenal": "git+https://github.com/scality/arsenal#8.1.130",
+    "arsenal": "git+https://github.com/scality/arsenal#1f20fe0fdbc0aa55ef71a4c870cd9230f75d494d",
```
Comment (benzekrimaha) [inline package.json]: The project version will be bumped once arsn released

File: lib/api/apiUtils/object/createAndStoreObject.js [hunk 2]
```diff
@@ -193,6 +193,10 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         metadataStoreParams.oldReplayId = objMD.uploadId;
     }
 
+    if (objMD && !bucketMD.isVersioningEnabled() && objMD.archive && Object.keys(objMD.archive).length > 0) {
```
Comment (francoisferrand) [inline lib/api/apiUtils/object/createAndStoreObject.js]: * This handles the case of bucket without versioning ; we may however have the same issue in version suspended case ? @Kerkesni, is there maybe already a function to check if we are in these kind of condition?
* no reason to check Object.keys(objMD.archive) : we should simply check if the object has been written to "cold", i.e. `objMD?.archive?.archiveInfo` (and you must not check the content of `archiveInfo`, it is supposed to stay opaque: hence the type `any` in arsenal's `ObjectMDArchive`)

AFAIR, race conditions (e.g. object deleted/replaced/...) during the archival process should be handled already when finishing the transition : if the object does not exist (or etag does not match), backbeat will post a Delete/GC request back to tape I think...

**Review by francoisferrand — APPROVED** [review]

File: lib/api/apiUtils/object/createAndStoreObject.js (hunk 2 above)
Comment (welansari) [reply lib/api/apiUtils/object/createAndStoreObject.js]: Looking at the function  `isVersioningEnabled`
```
    isVersioningEnabled() {
        const versioningConfig = this.getVersioningConfiguration();
        return versioningConfig ? versioningConfig.Status === 'Enabled' : false;
    }
```
This should handle the versioning suspended case also. (returns false when the bucket is non versioned and when it's in versioning suspended mode)

File: package.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
   "dependencies": {
     "@azure/storage-blob": "^12.12.0",
     "@hapi/joi": "^17.1.0",
-    "arsenal": "git+https://github.com/scality/arsenal#8.1.130",
+    "arsenal": "git+https://github.com/scality/arsenal#817bb836ec0e04fbf083b4c57e42e3986dcdc9bc",
```
Comment (ghost) [inline package.json]: Just a reminder to bump before merging

**Review by ghost — APPROVED** [review]

**Pull request merged**